### PR TITLE
Add rule snapshots and robustify Prettier rule validation

### PR DESCRIPTION
**Pull request by rekmarks**
This PR adds a validation script run on `yarn lint` that accomplishes 2 things:

1. It ensures that neither we nor any config that we extend enables any Prettier rules that should be disabled. Previously, we only checked our own rules, and not those of configs that we extend.
2. Creates rule snapshots to make it easy to understand the impact of any changes we make to our configs, such as changing our specified rules or the configs that we extend.

If the script is in write mode (by being given the argument `--write` or `-w`), it will overwrite the existing rule snapshots. Otherwise, it will compare the computed snapshot to the snapshot stored on disk, and exit with an error if they aren't equal. `yarn lint` runs the script in validation mode while `yarn lint:fix` runs it in write mode.

This PR will be followed up by one addition to this script, which will ensure that we only enumerate rules that are configured differently from configs that we extend. This will make it a breeze to extend other configs without enumerating, maintaining, and understanding every single rule ourselves.

This PR was made possible by [@eslint/eslintrc](https://www.npmjs.com/package/@eslint/eslintrc).

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -7,11 +7,13 @@ module.exports = {
     {
       files: ['./scripts/*.js'],
       parserOptions: {
+        ecmaVersion: 2018,
```
Comment (rekmarks) [inline .eslintrc.js:10]: Node 12 has 100% ES2018 support per https://node.green.

File: packages/base/rules-snapshot.json [hunk 2]
```diff
@@ -0,0 +1,503 @@
+{
+  "@typescript-eslint/brace-style": "off",
```
Comment (brad-decker) [inline packages/base/rules-snapshot.json:2]: are all rules listed in each package? Is that for completeness to ensure that if we accidently, for example, add the typescript package into the base that we'll see it flipped on? Would it be better to only list the ones specific for this package and see that in snapshots something is added? I probably just don't understand the method of how this is generated yet.

Comment (brad-decker) [reply packages/base/rules-snapshot.json:2]: or is it that prettier turns all these rules off! (ding ding ding) nvm me

**Review by brad-decker — APPROVED** [review]
okay this is dope. Having safety when making changes to know exactly what the impact of a change will be is great.

**Comment by Gudahtt** [comment]
This looks fantastic!

**Pull request merged**